docs/proposed/psa-driver-interface.md

```diff
@@ -600,13 +600,11 @@ This concept, or some other way to reuse portable code such as specifying inner
 
 How does `psa_key_derivation_output_key` work when the extraction part and the expansion part use different drivers?
 
-#### Public key not stored
+#### Public key calculation
 
-ECC key pairs are stored as the private key value only. The public key needs to be calculated from that.
+ECC key pairs are represented as the private key value only. The public key needs to be calculated from that. Both transparent drivers and opaque drivers provide a function to calculate the public key (`"export_public_key"`).
 
-It's fairly common for secure elements to also store only the private value. When a key is generated randomly or derived, the hardware reports the public key, and it is up to the software to store it. The current design makes this the job of the driver. Should it be the work of the core instead?
-
-Note that a solution also has to work for transparent keys, and when importing a private key into a secure element. If the core already has code to calculate the public key, it would make sense for the driver to be able to use it, especially in these cases.
+The specification doesn't mention when the public key might be calculated. The core may calculate it on creation, on demand, or anything in between. Opaque drivers have a choice of storing the public key in the key context or calculating it on demand and can convey whether the core should store the public key with the `"store_public_key"` property. Is this good enough or should the specification include non-functional requirements?
```

Comment:
Why would the core care about whether or not the opaque driver stores the public part in its key context? An opaque driver is free to declare context sizes, so as long as it declares the correct size, I don't see why the core should know about this behaviour through a property.

Reply:
The intent of the property is to allow the core to choose the size of the context based on the size of the key.

Reply:
Isn't that already supported through `key_pair_size`?

Reply:
`key_pair_size` is independent of the key size. See “Size of a statically allocated key context” below.

Reply:
`key_size_function` then? The reason that got added is to catch-all support all drivers that might err from the predefined path. I have the same comment about the symmetric key size multiplier, too.

You don't want to start adding flags for each and every custom usecase, that's why the key size function got added. Then why does store_public_key (and symmetric multiplier) deserve a flag?

Reply:
Indeed, if a driver uses `key_size_function`, the core won't know about its propensity to store the public key. The point of `store_public_key`, `key_pair_size` etc is to support drivers without using `key_size_function`, for systems that don't allocate arbitrarily-sized memory dynamically. I tried to have enough granularity to allow a reasonable compromise between complexity and wasted memory. For example it's possible to have a fixed array of slots, but with slots of different sizes to accommodate keys of different sizes.

Reply:
Then I suggest adding a size multiplier instead. Even if a driver indicates it wants to store the public key, you wouldn't know whether it wants to store in compressed or uncompressed form. Plus, the multiplier would create symmetry with symmetric key handling.